### DISCUSSION TITLE
Is it too late to start now

### DISCUSSION THREAD

**Discussion by mike-uk** [GerryDazoo/Slinger#560] [Q&A] [answered]
Is it too late to start now to get this working on a pro HD sb300 box since the server has already shutdown? Is it possible still to get the admin password and update firmware? Do I also have to block the outgoing port 5678 so it wont get bricked. If the unit is already bricked it can still be hard reset to start again? I have a pi ready to go. My relative have an unwanted unit also I see some of auction site.

**Comment by bradut** [comment, marked as answer]
No, for a SlingBox PRO-HD it is not too late. 
I have bricked and unbricked mine multiple times. 
Thanks to Gerry, I learned that by pressing the reset button for ~60 seconds, the PRO-HD will be reverted to its factory "admin" password (but will keep the latest firmware in case you've updated).

Here (or on Discord) you will also find the links to the latest firmware and examples about blocking the outbound port 5678.

Please search (here or on Discord) for the PDF doc that explains step by step how to resuscitate.
I think it is [this one.](https://www.mediafire.com/file/ppw6ep09t99xhu0/Slingbox_ProHD_PC_Unbricking-Setup-Network-Guide.pdf/file)
I will mention that, despite the PDF doc says that you cannot update the analog channels, actually you can. In some occasions I had to restart SlingPlayer 1.5 after setup, in order to be able to stream the video

**Reply by mike-uk** [reply]
Thank you for your reply, I have just ordered a used box hopefully get this working soon. Is it only port 5678 I need to block or 80,443,3074,3490,3724,5678,5680,8080?

**Reply by GerryDazoo** [reply]
It's just port 5678. That long list was from when we were in panic mode and we blocked everything going anywhere near Mr. Sling. Since then we've narrowed it down to port 5678. In my case I've blocked all outgoing traffic from my slingbox based on it's MAC address.

**Reply by ChrizK-oldGit** [reply]
Would you mind confirming the same for a slingbox M1?
To be honest, I am only starting to get my head around this workaround.  I initially thought that Slinger is a firmware overwrite on the slingbox, but would I be correct that it is a perl program (?) that runs to 'replace' the sling servers (performing as a media player)?  If that is the case, could I run Slinger on my Synology NAS ... or have I got the solution completely wrong?
(My M1 has been in the cupboard for years, but I recently subscribed to Virgin Media which I would like to view 'in the back room' where there is no easy way to run the Virgin cable to, and relocate the box ...why 'they' think there is no market for this is beyond me.  I guess they simply didn't sell enough...)

**Reply by GerryDazoo** [reply]
As far as we know the 500/M1 have been unscathed but we're recommending blocking outgoing Internet access from the Slingbox Just In Case.
It's a Python program and runs fine on a Synology NAS. My server is running on my DS218. It is not a media player. It servers out media streams to your favourite media player.

**Reply by ChrizK-oldGit** [reply]
> As far as we know the 500/M1 have been unscathed but we're recommending blocking outgoing Internet access from the Slingbox Just In Case. It's a Python program and runs fine on a Synology NAS. My server is running on my DS218. It is not a media player. It servers out media streams to your favourite media player.

Thanks Gerry, is there a link to an article telling me which addresses (mask?) to block?

You have slightly confused me though (my lack of understanding of the situation).
Should I install a Slinger server in my network (on NAS), or should the M1 simply work when I turn it on?

TBH it has been so long since I last used it, I can't even remember 'how'.
By 'work', can I install a media player on a Smart TV/Firestick/Roku and it will identify the Slingplayer as a source (maybe DLNA ...or is that just for files ...)?
I really apologise for my dumb questions.  I will be starting at ground zero  ... I assume I need to buy something like the folowing for the Virgin box ... https://www.amazon.co.uk/dp/B07VGWL5P1/?coliid=I3UC1YSJEUB7V7&colid=1OCDT9I5GVZV4&psc=1&ref_=lv_ov_lig_dp_it

**Reply by mike-uk** [reply]
I think its out going 5678 port that is need to block. I block the whole port in/out but still waiting for the pro/hd port to come.

The hd component goes into slingbox. The slinger is installed on a windows pc or linux box like raspberry pi (save power). You connect to the slinger server to watch not the slingbox.

Does that converter box support 1080i instead of p? I see something like that from aliexpress £10 + vat

**Reply by ChrizK-oldGit** [reply]
Oh, blocking by port ...did I miss this in the README?

I will be using linux (Synology), other than slingbox_server.py, config.ini and sling.service, do I need anything else (I understand I need 'flask' and 'netifaces' Python modules)?

£10 ... I would wonder if it is any good!
I asked the question on Amazon and someone kindly replied (with a proper answer!)
Q: 'not support 4k signal' - does it negotiate with input device? (i can plug virgin tv in non-4k (hd)tv and it works, but i believe it is 4k capable)
A: The instruction book says, maximum resolution is 1080p...on the outside of the box it states,
"Supports high res video up to 1080p and 4k2k (4k2k products)" your guess is as good as mine, doesn't make a lot of sense, I'm sorry but I haven't got a device with 4k out to test it out.I would be surprised if it was 4k capable. My laptop outputs 1920 by 1080, it works perfectly with that.

**Reply by mike-uk** [reply]
Can you not switch the 4k feed to hd on your virgin box?  I think only iplayer and youtube support 4k at the moment. The pro-hd model has tv tunner so I am not plugging in anything. There is another pro-hd box I want to set up in asia and the digital tuner doesnt work over there so will need to get a old box that support it or get converter. I also read that only 1080i would work not 1080p on pro-hd. Have you got the M1 box already?

This one on ebay seem to convert also
https://www.ebay.co.uk/itm/334420922681?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=mr3bhlyurqy&sssrc=2349624&ssuid=foH8X5k8QVi&var=&widget_ver=artemis&media=COPY

**Reply by ChrizK-oldGit** [reply]
> Can you not switch the 4k feed to hd on your virgin box?
Have you got the M1 box already?
> 
> Link to the box YPbPr R/L to HDMI-Compatible Converter 1080P Component Video Converter Audio Adapter Splitter for DVD HDTV Monitor Projector https://a.aliexpress.com/_EHJbrWf
> 
> This one on ebay seem to convert also https://www.ebay.co.uk/itm/334420922681?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=mr3bhlyurqy&sssrc=2349624&ssuid=foH8X5k8QVi&var=&widget_ver=artemis&media=COPY

Thanks for the links ...they are interesting options.  I haven't seen any method of switching to HD.  Given the severe limitations on the Virgin Hub, I doubt they give any real control in their TV box :)
My setup is probably going to be quite convoluted and probably negates 4K anyway ... I will firstly run to an HDMI splitter so I can feed the TV and also the RGB converter > Slingbox.
(I have literally only dug up the M1 out of the cupboard before googling to find the problems with the service, the Slinger workaround, the required converters, the need for something to run the server, being my Synology NAS ...with the potential of learning how to use Docker ...this is becoming quite a journey!)

**Reply by mike-uk** [reply]
You can set output format from the box as below for v6 box then you will only have to deal with hdmi. I think I will stream 720p anyway let alone 4k. I think all the setting is in the config files. Have to wait for my box to come, I am starting with orange pi zero.  May be its easier just to buy a hd or hd-pro box with dvb-t tuner than get all the convertor etc. They are as low as £10 i see as its just a brick if you dont know linux. But still need convertor for virgin box.

Sorry the aliexpress box is wrong one you want hdmi in and yPbPr out not the other way round. The ebay one is correct.

------------
Press Home on your Virgin TV remote
Go to Help & Settings
Click on Settings
Select Video Output
Go to Video Output format

**Reply by ChrizK-oldGit** [reply]
> I think its out going 5678 port that is need to block. I block the whole port in/out but still waiting for the pro/hd port to come.

Oh no, I can't see anywhere to block a port on Virgin Hub 4.  It has **IP and Port Filtering**, but there is nowhere to add a Port.
If I choose to add a rule, I am only prompted for Source and Destination IP address (All, Single, Range). I can't use ip:port as the input box length limited.  It is really annoying that if I specify an IP and add the rule, the resulting display shows Source Port and Destination Port as All, but no way to edit them :(

I would really appreciate knowing how big a problem this may be.  I haven't even dared power the M1 on yet!!!

Actually, re-reading @GerryDazoo 's advice, and @mike-uk mention of MAC address ... Can I simply block the MAC address for ALL internet traffic?

**Reply by mike-uk** [reply]
My virgin hub 3 is in modem mode so I can't check. I use another router with DD-wrt firmware, virgin hub sucks. 
Found a video about hub3 setting not sure if its the same for hub 4. You can reserve an IP address with mac address, I think then go to port filter instead of port forwarding.

https://youtu.be/yz9O0hOBfiI?t=186

The slingbox don't need the internet if your server is local but it doesn't need to be able to work on the local network. I am still waiting my pro had box, Royal mail had it since Monday, I am setting up my orange pi right now.

**Reply by ChrizK-oldGit** [reply]
Thank you @mike-uk , appreciate the effort you went to.  Unfortunately I am just starting to understand that this solution appears to rely on knowing the sling media account information.  As mentioned, the M1 has been in a cupboard for 'years'.  I don't even remember using it ... I remember buying it at Costco thinking it would be useful, but that is about it.  Can't find any emails from Slingmedia confirming my account or even sending me spam ...I wonder if I ever plugged it in ...perhaps that means it is 'default' and the password is 'admin'.  Had my Amazon delivery today, so will try and hook up the Virgin V6 tomorrow and spin up the Docker.
Would you know what sbtype I should use for the M1?

**Reply by mike-uk** [reply]
I wish i can help you but as I haven't setup myself I can't. May be go on discord it seem more active on there. You can also search the old chat.

Dont think you need to do anything to sbtype it is default "350/500/M1/M2". There are 2 config the other is for more than one box and with out the comments, may be look at that will give you idea what line to use.

I think few things missing from read me file.

**Reply by GerryDazoo** [reply]
chris, If the box was ever on line, the encrypted password for it may be buried in the SlingMedia log files on your PC. Search the discussions for retrieving the password. Did you try the default password "admin"? 
Worst case you can do a factory reset but that will wipe out any configuration that was downloaded for the IR remote control. But if it has never been setup you're no worse off.

**Reply by ChrizK-oldGit** [reply]
Thanks guys, that sounds much more promising than I was anticipating.  I do remember the M1 sitting on a shelf with a red light, and the IR dangling.  The face is certainly scratched indicating I did use it at some point.  That all said, I would have attempted to use it with a Sky  box and may have stopped using it when Sky HD was brought in, with no RGB/YPbPr.  Now my need is for Virgin V6, so I guess any IR remote would not be suitable anyway.
(I have certainly changed laptops several times, and also upgraded to Windows 10 ...but perhaps my old Dell has something ...if it boots ..)

Although I suspect I will have to reset in the hope to use 'admin', the password thing intrigues me. I had in my head that using the APIs to secure.sling.com and logging in with an existing account, would communicate back to the Slingbox and retrieve the password.  Given that we should block outgoing access, I guess this isn't the case, and perhaps the APIs retrieves the password from a Sling Media database...is that the case?  Of course, that means the steps will only work as long as Sling Media keep the database running/populated.

I will continue with some renewed hope ... thanks again

**Reply by GerryDazoo** [reply]
As far as we can tell, there is no way to recover the password from the box. That would be a huge security hole. If you don't have your Sling Account info or can't find the encrypted password in the log files you have no choice but to do a factory reset. But it sounds like the remote info loaded into the Slingbox is of no use so you won't be any worse off.

**Reply by ChrizK-oldGit** [reply]
Again, thanks for the advice.
Starting up the Docker container allowing the server to find the Slingplayer provided the name I gave it, which indicated the email address.  I then installed Slingplayer with the hope to use the method suggested by [socelillyhammer](https://github.com/socelillyhammer) in [HOW TO FIND THE PW TO 500/350/M1/M2](https://github.com/GerryDazoo/Slinger/discussions/374).  Unfortunately, no matter whatever I guessed as my 'normal passwords' failed, but trying to create a new account with the same email address confirmed I used it.

Anyway, I appreciate the advice to bite the bullet and reset.
I also found your comment about IR controllers interesting.  If I am correct, I believe the situation is that the community can only use an IR controller that was **installed when the service was active**, and the customisation that is discussed on this forum is to provide 'skins' to be used with the (already installed) IR controller.  That being the case, those changing the source they connect to will lose the IR controller function (on the basis that it is not compatible with the previously used device).
(hoping others find my journey informative as it may help with the general understanding of the problem and solution)

**Reply by ChrizK-oldGit** [reply]
> chris, If the box was ever on line, the encrypted password for it may be buried in the SlingMedia log files on your PC. Search the discussions for retrieving the password. Did you try the default password "admin"? Worst case you can do a factory reset but that will wipe out any configuration that was downloaded for the IR remote control. But if it has never been setup you're no worse off.

Hi @GerryDazoo , as discussed, I 'think' I have reset the password (holding pin into reset button for 10 seconds while powered on).
Is it correct that I don't see any logged activity until a request to stream has been made?  Taking into your comment on VLC, I have used MX Player on my mobile (user icon > Network Stream >http://192.168.0.4:57120/slingbox > Play Now)

You will see below that the streaming request results in "Check your admin password".

Can you (anyone) confirm config.ini, **password=admin** is expected to be correct for Slingbox M1 ...and perhaps my reset was unsuccessful?

You will also see the warning about a hacker alert due to an invalid request.  Perhaps this is due to the incorrect password(?)
NOTE: I am still experimenting using the Docker installation, v3.08f (I will attempt 'raw' Synology later).  Also, the newly purchased HDMI>YbYr converter is really struggling with the Virgin V6.  Attaching a TV shows the picture is in Black and White.


Hacker Alert. Invalid Request from  ('192.168.0.195', 58936)
GET / HTTP/1.1
User-Agent: MXPlayer/1.61.4 (Linux; Android 13; en-GB; SM-S918B Build/TP1A.220624.014.S918BXXS1AWBM)
Host: 192.168.0.4:57120
Connection: Keep-Alive
Accept-Encoding: gzip
03/26/2023, 15:26:53.151   Streaming request from ('192.168.0.195', 58938)
Slinginfo  "350/500/M1/M2" 12 30 192.168.0.194 5301 3072 10
57120 Starting Stream for  192.168.0.195:58938
Connecting... ('192.168.0.194', 5301) Control
cmd: 0x67 err: 0x2 8
**57120 Error Starting Session. Check your admin password in config.ini file!**
57120 ERROR: Slingbox session startup failed.
57120 Logging Out

**Reply by ChrizK-oldGit** [reply]
> Can you (anyone) confirm config.ini, **password=admin** is expected to be correct for Slingbox M1 ...and perhaps my reset was unsuccessful?

I hit another reset and have moved forward.  I hope the community doesn't mind me recording my progress within the discussion, as I believe it may help others.
Seeing the logged output made a realise something obvious ...
`Slingbox Found 192.168.0.194 5301 **" My Slingbox "**`

The previous log still had the name I gave it when it was registered, so I hadn't reset it properly.
You will note below that I still get a hacker alert when  using MX Player from my mobile, however, it appears the password is accepted and a stream attempted.  Not sure what v4 fixes, but I think it unfair to comment further without moving onto the latest version.

```
Hacker Alert. Invalid Request from  ('192.168.0.195', 33248)
GET / HTTP/1.1
User-Agent: MXPlayer/1.61.4 (Linux; Android 13; en-GB; SM-S918B Build/TP1A.220624.014.S918BXXS1AWBM)
Host: 192.168.0.4:57120
Connection: Keep-Alive
Accept-Encoding: gzip
03/26/2023, 15:51:24.39   Streaming request from ('192.168.0.195', 33264)
Slinginfo  "350/500/M1/M2" 12 30 192.168.0.194 5301 3072 10
57120 Starting Stream for  192.168.0.195:33264
Connecting... ('192.168.0.194', 5301) Control
57120 Selecting Video Source 1
Keycodes= []
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.0.194', 5301) Stream
57120 Stream started at 03/26/2023, 15:51:26.480  0 0
03/26/2023, 15:51:26.480  57120 Stream Stopped Unexpectly. Kicked Off?
57120 Shutting down connections
57120 Logging Out
```

**Comment by cooley420** [comment]
I just want to let people know that if you are looking for your admin password you can download the slingplayer desktop app, open it, and sign in, since the slingbox servers are down it won't ever connect, but you can then go into your ~/appdata/roaming/sling media/slingplayer desktop/logs/console.log, and find the hashed password.  Search for password, and look for a 15 digit alpha numeric password.
